// simplebar-angular/esm2020/lib/simplebar-angular.module.mjs
import { NgModule } from '@angular/core';
import { SimplebarAngularComponent } from './simplebar-angular.component';
import * as i0 from "@angular/core";
export class SimplebarAngularModule {
}
/** @nocollapse */ SimplebarAngularModule.ɵfac = function SimplebarAngularModule_Factory(t) { return new (t || SimplebarAngularModule)(); };
/** @nocollapse */ SimplebarAngularModule.ɵmod = /** @pureOrBreakMyCode */ i0.ɵɵdefineNgModule({ type: SimplebarAngularModule });
/** @nocollapse */ SimplebarAngularModule.ɵinj = /** @pureOrBreakMyCode */ i0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(SimplebarAngularModule, [{
        type: NgModule,
        args: [{
                declarations: [SimplebarAngularComponent],
                imports: [],
                exports: [SimplebarAngularComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(SimplebarAngularModule, { declarations: [SimplebarAngularComponent], exports: [SimplebarAngularComponent] }); })();
